crypto/src/proofs/transparent.rs

```diff
@@ -458,52 +417,56 @@ impl SwapClaimProof {
         // * and verify the output notes
 
         // Swap NFT nullifier integrity. Ensure the nullifier is correctly formed.
-        if nullifier
-            != self
-                .nk
-                .derive_nullifier(position, &self.note_commitment_proof.commitment())
-        {
-            return Err(anyhow!("bad nullifier"));
-        }
+        gadgets::nullifier_integrity(
+            nullifier,
+            self.nk,
+            position,
+            self.note_commitment_proof.commitment(),
+        )?;
 
         let (lambda_1, lambda_2) = output_data.pro_rata_outputs((self.delta_1_i, self.delta_2_i));
```

Comment:
This is (implicitly) one of the proof statements -- we'll need to have a gadget that checks the pro rata outputs are correct.  This doesn't quite fit the exact same form as the other gadgets, though. Maybe we should add `lambda_1` and `lambda_2` to the transparent proof?  In that case, we could delay adding that gadget to a future PR, so that this one can be a pure rearrangement of the existing checks.